.github/workflows/rust.yml

```
@@ -36,6 +36,10 @@ jobs:
       run: rustup default ${{ matrix.rust_version }}
     - name: print rustc version
       run: rustc --version
+    # remove this step when MSRV >= 1.67.0
+    - name: degrade `time` for older toolchain
+      if: matrix.rust_version == '1.65.0' 
```

Given that you bumped rust version to 1.66, you have to update the condition as well:

```suggestion
    - name: downgrade `time` crate to support older Rust toolchain
      if: matrix.rust_version == '1.66.0' 
```